lib/std/sysrand.nim

```diff
@@ -0,0 +1,258 @@
+#
+#
+#            Nim's Runtime Library
+#        (c) Copyright 2021 Nim contributors
+#
+#    See the file "copying.txt", included in this
+#    distribution, for details about the copyright.
+#
+
+## Cryptographically secure pseudorandom number generator.
+## 
+## | Targets    | Implementation|
+## | :---         | ----:       |
+## | Windows| `BCryptGenRandom`_ |
+## | Linux| `getrandom`_ system call when available, otherwise `/dev/urandom`_ will be used|
+## | MacOSX| `getentropy`_ system call when available, otherwise `/dev/urandom`_ will be used|
+## | IOS  | `SecRandomCopyBytes`_|
+## | OpenBSD| `getentropy openbsd`_ system call when available, otherwise `/dev/urandom`_ will be used|
+## | FreeBSD| `getrandom freebsd`_ system call when available, otherwise `/dev/urandom`_ will be used|
+## | JS(Web Browser)| `getRandomValues`_|
+## | Other Unix platforms| `/dev/urandom`_|
+##
+## .. _BCryptGenRandom: https://docs.microsoft.com/en-us/windows/win32/api/bcrypt/nf-bcrypt-bcryptgenrandom
+## .. _getrandom: https://man7.org/linux/man-pages/man2/getrandom.2.html
+## .. _/dev/urandom: https://en.wikipedia.org/wiki//dev/random
+## .. _getentropy: https://www.unix.com/man-page/mojave/2/getentropy
+## .. _SecRandomCopyBytes: https://developer.apple.com/documentation/security/1399291-secrandomcopybytes?language=objc
+## .. _getentropy openbsd: https://man.openbsd.org/getentropy.2
+## .. _getrandom freebsd: https://www.freebsd.org/cgi/man.cgi?query=getrandom&manpath=FreeBSD+12.0-stable
+## .. _getRandomValues: https://www.w3.org/TR/WebCryptoAPI/#Crypto-method-getRandomValues
+##
+
+runnableExamples:
+  doAssert urandom(0).len == 0
+  doAssert urandom(113).len == 113
+  doAssert urandom(1234) != urandom(1234) # unlikely to fail in practice
+
+##
+## See also
+## ========
+## * `random module <random.html>`_
+## 
+
+import std/os
+
+
+
+when defined(posix):
```

Comment:
This has to come **after** ``defined(linux)`` etc!

Reply:
I don't understand.
The planned implementation is that first try `getrandom` in linux then fall back to read `/dev/urandom`.

Reply:
ah ok, the code is like this:
```nim
when defined(posix): ...

when defined(js):
elif defined(windows):
elif defined(linux):
elif ...
```
which seems fine